stacks-signer/src/main.rs

```diff
@@ -188,83 +96,160 @@ fn spawn_running_signer(
-        (
-            config.node_host,
-            config.stackerdb_contract_id,
-            config.stacks_private_key,
-        )
+fn handle_get_chunk(args: GetChunkArgs) {
+    debug!("Getting chunk...");
+    let mut session = stackerdb_session(args.db_args.host, args.db_args.contract);
+    let chunk_opt = session.get_chunk(args.slot_id, args.slot_version).unwrap();
+    write_chunk_to_stdout(chunk_opt);
+}
+
+fn handle_get_latest_chunk(args: GetLatestChunkArgs) {
+    debug!("Getting latest chunk...");
+    let mut session = stackerdb_session(args.db_args.host, args.db_args.contract);
+    let chunk_opt = session.get_latest_chunk(args.slot_id).unwrap();
+    write_chunk_to_stdout(chunk_opt);
+}
+
+fn handle_list_chunks(args: StackerDBArgs) {
+    debug!("Listing chunks...");
+    let mut session = stackerdb_session(args.host, args.contract);
+    let chunk_list = session.list_chunks().unwrap();
+    println!("{}", serde_json::to_string(&chunk_list).unwrap());
+}
+
+fn handle_put_chunk(args: PutChunkArgs) {
+    debug!("Putting chunk...");
+    let mut session = stackerdb_session(args.db_args.host, args.db_args.contract);
+    let mut chunk = StackerDBChunkData::new(args.slot_id, args.slot_version, args.data.clone());
+    chunk.sign(&args.private_key).unwrap();
+    let chunk_ack = session.put_chunk(chunk).unwrap();
+    println!("{}", serde_json::to_string(&chunk_ack).unwrap());
+}
+
+fn handle_generate_files(args: GenerateFilesArgs) {
+    debug!("Generating files...");
+    let signer_stacks_private_keys = if let Some(path) = args.private_keys {
+        let file = File::open(&path).unwrap();
+        let reader = io::BufReader::new(file);
+
+        let private_keys: Vec<String> = reader.lines().collect::<Result<_, _>>().unwrap();
+        println!("{}", StacksPrivateKey::new().to_hex());
+        let private_keys = private_keys
+            .iter()
+            .map(|key| StacksPrivateKey::from_hex(key).expect("Failed to parse private key."))
+            .collect::<Vec<StacksPrivateKey>>();
+        if private_keys.is_empty() {
+            panic!("Private keys file is empty.");
+        }
+        private_keys
     } else {
-        (
-            cli.host.unwrap(),
-            cli.contract.unwrap(),
-            cli.private_key.unwrap(),
-        )
+        let num_signers = args.num_signers.unwrap();
+        if num_signers == 0 {
+            panic!("--num-signers must be non-zero.");
+        }
+        (0..num_signers)
+            .map(|_| StacksPrivateKey::new())
+            .collect::<Vec<StacksPrivateKey>>()
     };
+    let signer_stacks_addresses = signer_stacks_private_keys
+        .iter()
+        .map(|key| to_addr(key, &args.network))
+        .collect::<Vec<StacksAddress>>();
+    // Build the stackerdb contract
+    let stackerdb_contract = build_stackerdb_contract(&signer_stacks_addresses);
+    debug!("Stacker DB Contract: {}", &stackerdb_contract);
+    let contract_id = QualifiedContractIdentifier::new(
```

Comment:
This implies that the first signer must be the one to submit the contract.  Is this intentional?  Would it make sense to just write the contract code to a file containing only the contract name?